pkg/systemdgen/systemdgen.go

```diff
@@ -109,6 +138,41 @@ func CreateContainerSystemdUnit(info *ContainerInfo, generateFiles bool) (string
 		info.Executable = executable
 	}
 
+	// Assemble the ExecStart command for generic units.
+	//
+	// Note that we cannot catch all corner cases here such that users
+	// *must* manually check the generated files.  A container might have
+	// been created via a Python script, which would certainly yield an
+	// invalid `info.CreateCommand`.  Hence, we're doing a best effort unit
+	// generation and don't try aiming at completeness.
+	if opts.Generic {
+		// The create command must at least have three arguments:
+		// 	/usr/bin/podman run $IMAGE
+		if len(info.CreateCommand) < 3 {
+			return "", errors.Errorf("container's create command is too short or invalid: %v", info.CreateCommand)
+		}
+		// We're hard-coding the first four arguments and append the
+		// CreatCommand with a stripped command and subcomand.
+		command := []string{
+			info.Executable,
+			"run",
+			"--conmon-pidfile", "/%t/%n-pid",
+			"--cidfile", "/%t/%n-cid",
+		}
+		// Check whether the container was created via
+		// container-{create,run} or the short form.
+		index := 2
+		if info.CreateCommand[1] == "container" {
+			index = 3
+			if len(info.CreateCommand) < 4 {
+				return "", errors.Errorf("container's create command is too short or invalid: %v", info.CreateCommand)
```

Comment:
duplication. Why not move this up, and combine with 151-153 into:
```go
    index := 2
    if len(info.CreateCommand) > 2 && info.CreateCommand[1] == "container" {
        index = 3
    }
    if len(info.CreateCommand) < index+1 {
        ...error too short
```

Reply:
Great idea!